snowfakery/data_generator_runtime.py

```diff
@@ -156,16 +211,40 @@ def register_intertable_reference(
             Dependency(table_name_from, table_name_to, fieldname)
         )
 
+    def reset_slots(self):
+        "At the beginning of every execution, reset the forward reference slots"
+        self.named_slots = {
+            name: NicknameSlot(table, self.id_manager)
+            for name, table in self.nicknames_and_tables.items()
+        }
+
+    def check_slots_filled(self):
+        not_filled = [
+            name
+            for name, slot in self.named_slots.items()
+            if slot.status == SlotState.ALLOCATED
+        ]
+        if len(not_filled) > 1:
+            raise DataGenNameError(
+                f"References not fulfilled: {','.join(not_filled)}", None, None
+            )
+        elif not_filled:
+            raise DataGenNameError(
+                f"Reference not fulfilled: {not_filled[0]}", None, None
+            )
```

Comment:
Why do you need a special case for length 1?

Reply:
7b57423